src/wildcard.c

```diff
@@ -13,34 +13,46 @@ match_t Wildcard_MatchChar(const char *pattern, size_t p_len, const char *str, s
   const char *ns_itr = NULL;
   int i = 0;
   while (++i) {
-    //printf("%d ", i);
-    if (pattern_end != pattern_itr) {
+    if (pattern_end > pattern_itr) {
       const char c = *pattern_itr;
-      if ((str_end != str_itr) && (c == *str_itr || c == '?')) {
+      if ((str_end > str_itr) && (c == *str_itr || c == '?') && (c != '*')) {
+        // Equal characters or '?' match --> advance both pointers
         ++str_itr;
         ++pattern_itr;
         continue;
       } else if (c == '*') {
-        while ((pattern_end != pattern_itr) && (*pattern_itr == '*')) {
+        while ((pattern_end > pattern_itr) && (*pattern_itr == '*')) {
+          // Multiple '*' are equivalent to a single '*' --> skip them
           ++pattern_itr;
         }
         const char d = *pattern_itr;
-        while ((str_end != str_itr) && !(d == *str_itr || d == '?')) {
-          ++str_itr;
+        if (d != '?') {
```

Comment:
maybe also check that the pattern iterator didn't reach the end?

Reply:
Why? `\0` is a character as well..